File: inhouse/simple_16mer.py
# Old 16-mer experiment (Python)
# - Sums up the output of "process" function (here just returns k-mer length) 
#   on each 16-mers (step 2) in the file and outputs sum 
#   and total number of processed k-mers

import sys

def process(k):
   return len(k)

def ksplit(s, k, step):
   i = 0
   while i + k < len(s):
      yield s[i:i + k]
      i += step

f = open(sys.argv[1], 'r')
total, t2 = 0, 0
k, step = 16, 2
for l in f:
   for s in ksplit(l.strip(), k, step):
      total += process(s)
      t2 += 1
print total, t2

File: inhouse/simple_16mer.jl
# Old 16-mer experiment (Julia)
# - Sums up the output of "process" function (here just returns k-mer length) 
#   on each 16-mers (step 2) in the file and outputs sum 
#   and total number of processed k-mers

function process(s)
	length(s)
end

function main(arg)
	f = open(arg)
	t1, t2 = 0, 0
	for l in eachline(f)
		k, stp = 16, 2
	  i = 1
	  while i + k <= length(l)
	    sk = l[i:i + k - 1]
	    t1 += process(sk)
	   	t2 += 1
	    i += stp
	  end
	end
	println("$t1 $t2")
end

main(ARGS[1])